ckanext/s3filestore/uploader.py

```diff
@@ -308,7 +327,7 @@ def get_signed_url_to_key(self, key, extra_params={}):
         if is_public_read:
             url = url.split('?')[0] + '?ETag=' + metadata['ETag']
 
-        self._cache_put(key, url)
+        self._cache_put_url(key, url)
```

Comment:
can you look at doing a refactor on this method ```def get_signed_url_to_key(self, key, extra_params={}):```

to check cache at the start before doing the s3 head calls etc.


We also saw that if redis/cache_put fails, it stops the url being returned to the customer. Can you wrap the redis put with a try, log error continue as its a nice to have not a show stopper for getting the url.

Also on redis get, do similar that if it throws exception that None is return per it not finding a cached key.